### DISCUSSION TITLE
这个算不算 Bug

### DISCUSSION THREAD

**Discussion by wuliaodexiaoluo** [Meloong-Git/PCL#4630] [提问] [answered]
## 复现方法:
给官启目录挂软链
然后复制一份版本文件夹，PCL 识别到错误的文件夹，在版本列表内显示卡片后清空这个错误的版本文件夹，此时回到 PCL，没挂软链的目录正常，挂了的继续报错
此时看起来完全符合 #3812 中龙猫对于刻意触发的定义，并且可能符合预期结果：因为特殊的文件环境导致误判，而正常的则不会误判
## 此时就和龙猫所定义的 Bug 有逻辑上的冲突
如果以不符合预期的行为即是 Bug 的定义，并且结合大部分行为符合预期即为正常环境（其他功能是正常的）来看，这确实是一个 Bug
但是如果以刻意触发的定义来看，这并非是一个Bug

## 如果没有优先级，比如不符合预期的优先级大于环境，或者环境的优先级大于不符合预期，那么这个 Bug 就成了薛定谔的猫，处于既是 Bug 又不是 Bug 的叠加状态
如果获取文件列表时，获取失败就不会作为空文件夹对待的话，那么这个解释就和上文的复现步骤与结果冲突，如果获取失败就作为空文件夹并跳过 JSON 检查，那么全部版本都应该被作为空文件夹对待，不可能显示为错误的版本，如果同时检查，获取不到文件列表自然就不可能获取文件夹列表，也就不可能拼接成 JSON 文件路径

**Comment by youzi-3222** [comment]
问题已复现，其原因应为 `软链` 的名称与 `软链所指向的目录下 Json 文件` 的名称不同，应当不属于预期行为（除非给 Json 文件也弄个软链）。

---

另外，个人认为对于一些比较擦边的问题，没必要太纠结到底算不算 bug。引用 #3812：

> 这个定义的 **唯一目的** 是阻止 “为了获取活跃橙，而刻意地去构造一些不正常的情况” 这一滥用行为的。

**Comment by WForst-Breeze** [comment, marked as answer]
事实上，个人认为，不需要太过纠结所谓到底属不属于「刻意触发」，这本身也是开发者的 **主观判断**，尝试以 **客观标准** 衡量会搞得一头雾水。你我都不会读心术，也不清楚一些问题的出现到底是预期内的结果还是意外发生的错误，作为反馈者纠结这个其实莫得太大必要。

可以详细描述问题，直接交个 Issue，等待龙猫判断就好了 👀
